pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -410,6 +412,12 @@ func CreateGCECloud(config *CloudConfig) (*GCECloud, error) {
 		kmsConfig:                config.KMSConfig,
 	}
 
+	// Register the provided KMS service with the plugin registry to allow using
+	// it for encryption of resources.
+	encryptionconfig.KMSPluginRegistry.Register(KMSServiceName, func(_ io.Reader) (envelope.Service, error) {
```

Comment:
We are registering a constructor for the plugin. Even if the plugin is not stateless, the constructor would be stateless (if written like this one).
I'll make sure the registry does not throw errors for repeated registrations of a plugin.